libvast/vast/system/reader_command.hpp

```diff
@@ -117,6 +117,8 @@ caf::message reader_command(const command& cmd, caf::actor_system& sys,
       }
     }
     Reader reader{slice_type};
+    if (schema)
+      reader.schema(*schema);
```

Comment:
In case we don't have a schema supplied via `--schema` or `--schema-file`, `schema` is a pointer onto the global `event_types`. Maybe `Reader::schema(vast::schema)` should get a `const&` overload that we can use for the unmodified variant instead?

Reply:
In this case, I'd say let's go for value semantics. A schema is a list of `type`s and they have reference semantics. We're also not on the critical path.